packages/cozy-viewer/src/providers/ShareModalProvider.jsx

```diff
@@ -0,0 +1,41 @@
+import React, { useMemo, useContext, useState } from 'react'
+
+import { ShareModal } from 'cozy-sharing'
+
+export const ShareModalContext = React.createContext()
+
+export const useShareModal = () => {
+  const context = useContext(ShareModalContext)
+
+  if (!context) {
+    throw new Error('useShareModal must be used within a ShareModalProvider')
+  }
+  return context
+}
+
+const ShareModalProvider = ({ file, children }) => {
+  const [showShareModal, setShowShareModal] = useState(false)
```

Comment:
This approach via a new Provider will cause a new rendering of all its children each time the ShareModal is opened and closed.

Reply:
Oui vu ensemble, la modale de partage est maintenant utilisé à différents endroits, c'est le prix à payer. A voir à l'usage si on peut optimiser 👍